### PR TITLE
SAML: add securitymanager property

### PR DESCRIPTION
**Pull request by tduehr**
Adds a property to control the security manager class for opensaml in the cas-server-support-saml package. This is used to support jdk8 by setting it to `com.sun.org.apache.xerces.internal.util.XMLSecurityManager` or to use Xerces with `org.apache.xerces.util.SecurityManager`

### CODE REVIEW

File: cas-server-support-saml/src/main/resources/META-INF/spring/opensaml-config.xml [hunk 1]
```diff
@@ -47,9 +47,11 @@
           init-method="initialize">
         <property name="builderAttributes">
             <map>
-                <!-- Sun/Oracle is the default, for Xerces, set property to org.apache.xerces.util.SecurityManager -->
+                <!-- Sun/Oracle is the default, for Xerces, set property to org.apache.xerces.util.SecurityManager
```
Comment (mmoayyed) [inline cas-server-support-saml/src/main/resources/META-INF/spring/opensaml-config.xml]: Remove the above comments from here. Add the property to cas.properties, and explain what the setting does. Comment out the setting of course.

**Comment by mmoayyed** [comment]
:+1: with a comment on organization of comments. 

Also, please update the documentation.

**Comment by tduehr** [comment]
Done

**Comment by mmoayyed** [comment]
Perfect! Thank you.

**Comment by mmoayyed** [comment]
One last thing, I do still see doc updates missing from this PR. You'll find the docs under the documentation module in master. Please update those as well to make this new setting known.

**Comment by tduehr** [comment]
I'd just forgotten to actually commit it before pushing...

**Comment by mmoayyed** [comment]
Thanks again. I marked this for 4.2, given that we are now in RC mode and this isnt really a bug as we don't support JDK8 yet officially. Once 4.1 is out, this could go into master, and 4.1.x.

**Comment by mmoayyed** [comment]
replaces https://github.com/Jasig/cas/pull/1120

**Comment by tduehr** [comment]
I'm not entirely sure XMLSecurityManager is needed for jdk8. I do know I get a cast exception without changing it to `org.apache.xerces.util.SecurityManager`. I get a ClassNotFound exception for XMLSecurityManager.

**Comment by mmoayyed** [comment]
What is the Java version you run?

**Comment by tduehr** [comment]
openjdk8 on ubuntu 15.04

**Comment by mmoayyed** [comment]
OK, so that's where you ClassNotFound issue comes in. We don't officially support Java 8, and we also don't support OpenJDK. In your dependency list, you probably have xerces pulled in somehow which causes the cast issue.

**Comment by tduehr** [comment]
Xerces is being pulled in by the `cas-server-support-openid`.

The relevant section of `mvn dependency:tree`

```
[INFO] +- org.jasig.cas:cas-server-support-openid:jar:4.1.0-RC3-SNAPSHOT:compile
[INFO] |  +- org.openid4java:openid4java:jar:0.9.8:compile
[INFO] |  |  +- commons-logging:commons-logging:jar:1.1.1:compile
[INFO] |  |  +- net.sourceforge.nekohtml:nekohtml:jar:1.9.10:compile
[INFO] |  |  +- com.google.inject:guice:jar:2.0:compile
[INFO] |  |  \- xerces:xercesImpl:jar:2.8.1:compile
[INFO] |  \- org.apache.httpcomponents:httpclient:jar:4.4.1:compile
[INFO] |     \- org.apache.httpcomponents:httpcore:jar:4.4.1:compile
```

**Comment by mmoayyed** [comment]
If is possible to exclude xerces from that openid module? Would that cause issues? If not, we could directly exclude it in source.

**Comment by tduehr** [comment]
It may be. I'll try when I get a chance today.

**Comment by mmoayyed** [comment]
Great. If you are successful, please issue a separate PR. Thanks again.

**Comment by tduehr** [comment]
It works with xerces excluded. I will send a PR to exclude that dependency instead.

**Comment by mmoayyed** [comment]
Much appreciated. That PR, we can include in 4.1.

**Pull request closed without merging**